docs/posts/pydata-performance-part2/index.qmd

```diff
@@ -0,0 +1,283 @@
+about the same runtime performance, and lag far behind DuckDB on this workload.
+There's negligible performance difference between Ibis and the backend native
+APIs.
+
+## Motivation
+
+This is part 2 of a series of posts showing performance across various backends
+that Ibis supports.
+
+Check out [part 1](../pydata-performance/) if you haven't already!
+
+In this post, I'll continue with the [Polars](../../backends/polars.qmd) and
+[DataFusion](../../backends/datafusion.qmd) backends.
+
+I show each tool using both the Ibis API and the tool's native API. We'll see
+that the performance difference between these approaches is negligible.
+
+```{python}
+#| echo: false
+def show_file(path, language: str = "python") -> None:
+    with open(path) as f:
+        source = f.read()
+    print(f"```{language}\n{source}\n```")
+```
+
+## Setup
+
+I ran all of the code in this blog post on a machine with these specs.
+
+```{python}
+#| echo: false
+#| output: asis
+import os
+import platform
+import shutil
+
+import cpuinfo
+import psutil
+
+info = cpuinfo.get_cpu_info()
+uname = platform.uname()
+
+MiB = 1 << 20
+GiB = 1 << 30
+TiB = 1 << 40
+
+ram_gib = psutil.virtual_memory().total / GiB
+disk_tib = shutil.disk_usage("/").total / TiB
+
+lines = [
+    "| Component | Specification |",
+    "| --------- | ------------- |",
+    f"| CPU | {info['brand_raw']} ({os.cpu_count()} threads) |",
+    f"| RAM | {ram_gib:.0f} GiB |",
+    f"| Disk | {disk_tib:.1f} TiB SSD |",
+    f"| OS | NixOS ({uname.system} {uname.release}) |",
+]
+print("\n".join(lines))
+```
+
+### Library versions
+
+Here are the versions I used to run this experiment at the time of writing.
+
+```{python}
+#| echo: false
+#| output: asis
+import importlib
+import subprocess
+import sys
+
+import pandas as pd
+
+
+subprocess.run(("git", "fetch", "upstream"), check=True, capture_output=True)
+
+cmd = "git", "rev-parse", "--short", "upstream/master"
+proc = subprocess.run(cmd, check=True, text=True, capture_output=True)
+commit = proc.stdout.strip()
+link = f"https://github.com/ibis-project/ibis/tree/{commit}"
+
+version_pair = lambda name: (name, importlib.import_module(name).__version__)
+versions = pd.DataFrame(
+    [("Python", sys.version)] + sorted(
+        [
+            *map(version_pair, ("pandas", "polars", "datafusion", "pyarrow")),
+            ("ibis", f"[`{commit}`]({link})"),
+        ]
+    ),
+    columns=["Dependency", "Version"],
+)
+
+print(versions.to_markdown(index=False))
+```
+
+## Running the query across backends
+
+Here are the different Ibis expressions for each backend as well as the same
+query with native APIs, along with timed executions of the query.
+
+### DuckDB
+
+First, let's run the Ibis + DuckDB version of the query from the original post:
+
+```{python}
+#| echo: false
+#| output: asis
+show_file("./duckdb_ibis.py")
+```
+
+```{python}
+duckdb_ibis_results = %timeit -n1 -r1 -o %run duckdb_ibis.py
+df.head()
+```
+
+### DataFusion and Polars
```

Comment:
This is not showing up in the rendered html for some reason (which is very annoying)